### ISSUE TITLE
Invalid API test: '.set(key, value, ttl) sets a value that expires'

### ISSUE THREAD

**Issue by MySidesTheyAreGone** [lukechilds/keyv-test-suite#7]
I'm working on my own version of keyv-sql as we talked about a couple days ago. I'm running the tests and I believe in this one the ttl and delays you're setting are way too short. The test fails for me - let me get the code:

```js
	test.serial('.set(key, value, ttl) sets a value that expires', (() => {
		var _ref4 = _asyncToGenerator(function* (t) {
			const keyv = new Keyv({ store: store() });
			yield keyv.set('foo', 'bar', 100);
			t.is((yield keyv.get('foo')), 'bar');
			if (keyv.opts.store.ttlSupport === true) {
				yield (0, _delay2.default)(150);
			} else {
				_timekeeper2.default.freeze(Date.now() + 150);
			}
			t.is((yield keyv.get('foo')), undefined);
			_timekeeper2.default.reset();
		});

		return function (_x3) {
			return _ref4.apply(this, arguments);
		};
	})());
```

It fails specifically because if you set a 100 ms TTL:

	yield keyv.set('foo', 'bar', 100);

...then when you get the key right after:

	t.is((yield keyv.get('foo')), 'bar');

....the value has already expired, so I remove it from the db and return `undefined`. Printing out the expiration date and current time it's obvious that way more than 100 ms have passed since you first set the value:

```
1502530240656 <-- expiry
1502530240749 <-- now
key deleted, returning undef
```
So it takes around 200ms for the `keyv.get('foo'))` to get called.

I think the test is invalid because it passes by simply multiplying the various delays by 10. The following works:

```js
	test.serial('.set(key, value, ttl) sets a value that expires', (() => {
		var _ref4 = _asyncToGenerator(function* (t) {
			const keyv = new Keyv({ store: store() });
			yield keyv.set('foo', 'bar', 1000);
			t.is((yield keyv.get('foo')), 'bar');
			if (keyv.opts.store.ttlSupport === true) {
				yield (0, _delay2.default)(1500);
			} else {
				_timekeeper2.default.freeze(Date.now() + 1500);
			}
			t.is((yield keyv.get('foo')), undefined);
			_timekeeper2.default.reset();
		});

		return function (_x3) {
			return _ref4.apply(this, arguments);
		};
	})());
```

Are you setting a performance ceiling on purpose? Did I go crazy or something?

**Comment by lukechilds** [comment]
It's much easier to read the pre-transpiled test source btw ;)

https://github.com/lukechilds/keyv-test-suite/blob/master/src/api.js#L26

```js
test.serial('.set(key, value, ttl) sets a value that expires', async t => {
	const keyv = new Keyv({ store: store() });
	await keyv.set('foo', 'bar', 100);
	t.is(await keyv.get('foo'), 'bar');
	if (keyv.opts.store.ttlSupport === true) {
		await delay(150);
	} else {
		tk.freeze(Date.now() + 150);
	}
	t.is(await keyv.get('foo'), undefined);
	tk.reset();
});
```

OMG I could literally kiss you right now! I got stuck with this for ages, it was only failing intermittently and re-running them on Travis would always pass. I just automatically assumed it was the test checking the value had expired:

```js
t.is(await keyv.get('foo'), undefined);
```

that was failing because it was still set for some reason. I was pulling my hair out trying to figure out what the hell was going on. I played around with all sorts of mad delays between the expire checks to try and get rid of the errors.

The idea that it was

```js
await keyv.set('foo', 'bar', 100);
t.is(await keyv.get('foo'), 'bar');
```
which was actually failing never even crossed my mind. Your idea that after `await`ing the 100ms has already elapsed so the value no longer exists makes perfect sense!

> Are you setting a performance ceiling on purpose?

I was originally trying to keep the delays low just so the tests don't take ages to run.

Working on a fix now 👍

**Comment by MySidesTheyAreGone** [comment]
Yeah at the beginning I also assumed the second `get` was the problem. Know how I realized it was the first?

`console.log`

The one everybody tell you not to use.

Take THAT, everybody!!! `console.log` is awesome!!!1!

**Comment by lukechilds** [comment]
Normally AVA has really beautiful reporting and it'll print the line that fails but it doesn't seem to work when the tests imported as a dependency.

**Comment by lukechilds** [comment]
Are you able to test against the [`timings`](https://github.com/lukechilds/keyv-test-suite/tree/timings) branch and let me know if that resolves your issue?

**Comment by MySidesTheyAreGone** [comment]
It works! Thanks

**Comment by lukechilds** [comment]
@MySidesTheyAreGone No, thank you 😉👉

Just published @keyv/test-suite@1.5.2

`rm -rf node_modules && npm it`

**Comment by MySidesTheyAreGone** [comment]
Fastest debug in the history of mankind! Have a nice weekend
